network/game_test.go

```diff
@@ -0,0 +1,641 @@
+				ID:   `a`,
+				Name: `a`,
+			}, {
+				ID:   `b`,
+				Name: `b`,
+			}},
+			PlayerColors: map[model.PlayerID]string{
+				`a`: `blue`,
+				`b`: `red`,
+			},
+			BlockingPlayers: map[model.PlayerID]string{
+				`b`: `DealCards`,
+			},
+			CurrentDealer: `b`,
+		},
+	}}
+	for _, tc := range tests {
+		resp := NewCreateGameResponse(tc.game)
+		assert.Equal(t, tc.expResp, resp, tc.desc)
+	}
+}
+
+func TestNewGetGameResponse(t *testing.T) {
+	tests := []struct {
+		desc    string
+		game    model.Game
+		expResp GetGameResponse
+	}{{
+		desc: `shouldn't return hands or crib`,
+		game: model.Game{
+			ID: model.GameID(123456),
+			Players: []model.Player{{
+				ID:   `a`,
+				Name: `a`,
+			}, {
+				ID:   `b`,
+				Name: `b`,
+			}},
+			PlayerColors: map[model.PlayerID]model.PlayerColor{
+				`a`: model.Blue,
+				`b`: model.Red,
+			},
+			CurrentScores: map[model.PlayerColor]int{
+				model.Blue: 11,
+				model.Red:  22,
+			},
+			LagScores: map[model.PlayerColor]int{
+				model.Blue: 10,
+				model.Red:  20,
+			},
+			Phase: model.CribCounting,
+			BlockingPlayers: map[model.PlayerID]model.Blocker{
+				`b`: model.CountCrib,
+			},
+			CurrentDealer: `b`,
+			Hands: map[model.PlayerID][]model.Card{
+				`a`: modelCardsFromStrings(`ah`, `2h`, `3h`, `4h`),
+				`b`: modelCardsFromStrings(`as`, `2s`, `3s`, `4s`),
+			},
+			Crib:    modelCardsFromStrings(`5h`, `6h`, `5s`, `6s`),
+			CutCard: model.NewCardFromString(`5c`),
+			PeggedCards: []model.PeggedCard{{
+				Card:     model.NewCardFromString(`ah`),
+				Action:   0,
+				PlayerID: `a`,
+			}, {
+				Card:     model.NewCardFromString(`as`),
+				Action:   1,
+				PlayerID: `b`,
+			}, {
+				Card:     model.NewCardFromString(`2h`),
+				Action:   2,
+				PlayerID: `a`,
+			}, {
+				Card:     model.NewCardFromString(`2s`),
+				Action:   3,
+				PlayerID: `b`,
+			}, {
+				Card:     model.NewCardFromString(`3h`),
+				Action:   4,
+				PlayerID: `a`,
+			}, {
+				Card:     model.NewCardFromString(`3s`),
+				Action:   5,
+				PlayerID: `b`,
+			}, {
+				Card:     model.NewCardFromString(`4h`),
+				Action:   6,
+				PlayerID: `a`,
+			}, {
+				Card:     model.NewCardFromString(`4s`),
+				Action:   7,
+				PlayerID: `b`,
+			}},
+		},
+		expResp: GetGameResponse{
+			ID: model.GameID(123456),
+			Players: []Player{{
+				ID:   `a`,
+				Name: `a`,
+			}, {
+				ID:   `b`,
+				Name: `b`,
+			}},
+			PlayerColors: map[model.PlayerID]string{
+				`a`: `blue`,
+				`b`: `red`,
+			},
+			CurrentScores: map[string]int{
+				`blue`: 11,
+				`red`:  22,
+			},
+			LagScores: map[string]int{
+				`blue`: 10,
+				`red`:  20,
+			},
+			Phase: `CribCounting`,
+			BlockingPlayers: map[model.PlayerID]string{
+				`b`: `CountCrib`,
+			},
+			CurrentDealer: `b`,
+			Hands:         nil,
+			Crib:          nil,
+			CutCard: Card{
+				Suit:  `Clubs`,
+				Value: 5,
+				Name:  `5C`,
+			},
+			PeggedCards: cardsFromStrings(`AH`, `AS`, `2H`, `2S`, `3H`, `3S`, `4H`, `4S`),
```

Comment:
🔍 `PeggedCards` will also _need_ to identify who pegged them. We'll need a correlation to a playerID.